client/src/app/site/deployment-center/deployment-center-setup/step-configure/configure-vsts-build/functions-frameworks/functions-frameworks.component.ts

```diff
@@ -29,6 +30,21 @@ export class FunctionsFramworksComponent {
       value: WebAppFramework.PrecompiledFunction,
     },
   ];
+
+  private setupValidators() {
+    const workingDirectoryValidator = RegexValidator.create(
+      new RegExp(/^(\.{2,}|~|())\\(.)*$|^(\.{2,}|~|())\/(.)*$|(.)+:(.)*/),
```

Comment:
this is being repeated in multiple files. Any common where we can declare this?

Reply:
There's a constants file, where we can define these, but the pattern that's followed everywhere is keeping these variables local.. maybe we can move regex pattern to constants folder..